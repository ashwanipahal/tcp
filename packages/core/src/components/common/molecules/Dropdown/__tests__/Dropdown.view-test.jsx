import React from 'react';
import { shallow } from 'enzyme';
import DropdownVanilla from '../views';

describe('Dropdown component', () => {
  it('should renders correctly', () => {
    const props = {
      className: 'sample-class',
      options: {
        value: '1',
        content: (
          <div>
            <p>Test 1</p>
          </div>
        ),
        title: 'Test 1',
      },
      active: 'Active title',
      activeComponent: 'activeClassValue',
    };
    const component = shallow(<DropdownVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should renders correctly when optional fields are not present', () => {
    const props = {
      className: 'sample-class',
      options: {
        value: '1',
        content: (
          <div>
            <p>Test 1</p>
          </div>
        ),
        title: 'Test 1',
      },
    };
    const component = shallow(<DropdownVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
});
