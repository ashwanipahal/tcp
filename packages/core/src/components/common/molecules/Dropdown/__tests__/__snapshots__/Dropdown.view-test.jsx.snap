// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should renders componentDidUpdate  correctly 1`] = `
<Styled(BodyCopy)
  className="sample-class drop_down"
  color="text.primary"
  component="div"
  fontFamily="primary"
  fontSize={12}
  fontWeight="regular"
  letterSpacing="normal"
  lineHeight="normal"
  link={null}
  textAlign="left"
>
  <Styled(BodyCopy)
    className="customSelectTitle"
    color="text.primary"
    component="div"
    fontFamily="secondary"
    fontSize="fs14"
    fontWeight="extrabold"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    onClick={[Function]}
    textAlign="center"
  >
    <Styled(BodyCopy)
      className="customSelectArrow down"
      color="text.primary"
      component="div"
      fontFamily="secondary"
      fontSize={12}
      fontWeight="extrabold"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      textAlign="left"
    />
  </Styled(BodyCopy)>
</Styled(BodyCopy)>
`;

exports[`Dropdown component should renders componentDidUpdate with option length  correctly 1`] = `
<Styled(BodyCopy)
  className="sample-class drop_down"
  color="text.primary"
  component="div"
  fontFamily="primary"
  fontSize={12}
  fontWeight="regular"
  letterSpacing="normal"
  lineHeight="normal"
  link={null}
  textAlign="left"
>
  <Styled(BodyCopy)
    className="customSelectTitle"
    color="text.primary"
    component="div"
    fontFamily="secondary"
    fontSize="fs14"
    fontWeight="extrabold"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    onClick={[Function]}
    textAlign="center"
  >
    <Styled(BodyCopy)
      className="customSelectArrow down"
      color="text.primary"
      component="div"
      fontFamily="secondary"
      fontSize={12}
      fontWeight="extrabold"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      textAlign="left"
    />
  </Styled(BodyCopy)>
</Styled(BodyCopy)>
`;

exports[`Dropdown component should renders correctly 1`] = `
<Styled(BodyCopy)
  className="sample-class drop_down"
  color="text.primary"
  component="div"
  fontFamily="primary"
  fontSize={12}
  fontWeight="regular"
  letterSpacing="normal"
  lineHeight="normal"
  link={null}
  textAlign="left"
>
  <Styled(BodyCopy)
    className="customSelectTitle"
    color="text.primary"
    component="div"
    fontFamily="secondary"
    fontSize="fs14"
    fontWeight="extrabold"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    onClick={[Function]}
    textAlign="center"
  >
    <Styled(BodyCopy)
      className="customSelectArrow down"
      color="text.primary"
      component="div"
      fontFamily="secondary"
      fontSize={12}
      fontWeight="extrabold"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      textAlign="left"
    />
  </Styled(BodyCopy)>
</Styled(BodyCopy)>
`;

exports[`Dropdown component should renders correctly when optional fields are not present 1`] = `
<Styled(BodyCopy)
  className="sample-class drop_down"
  color="text.primary"
  component="div"
  fontFamily="primary"
  fontSize={12}
  fontWeight="regular"
  letterSpacing="normal"
  lineHeight="normal"
  link={null}
  textAlign="left"
>
  <Styled(BodyCopy)
    className="customSelectTitle"
    color="text.primary"
    component="div"
    fontFamily="secondary"
    fontSize="fs14"
    fontWeight="extrabold"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    onClick={[Function]}
    textAlign="center"
  >
    <Styled(BodyCopy)
      className="customSelectArrow down"
      color="text.primary"
      component="div"
      fontFamily="secondary"
      fontSize={12}
      fontWeight="extrabold"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      textAlign="left"
    />
  </Styled(BodyCopy)>
</Styled(BodyCopy)>
`;
